### PR TITLE
chrome_webgl2_workaround

### PR DESCRIPTION
**Pull request by juj**
Work around the issue that Chrome 56 and Chrome 57 shipped WebGL 2 without the new WebGL 2 function entry points. Fix a bug in glBufferData() where legacy GL emulation was used without gating for #if LEGACY_GL_EMULATION.

### CODE REVIEW

File: src/library_gl.js [hunk 1]
```diff
@@ -1507,17 +1513,24 @@ var LibraryGL = {
         usage = 0x88E8; // GL_DYNAMIC_DRAW
         break;
     }
+#endif
```
Comment (juj) [inline src/library_gl.js]: This block was actually missing `#if LEGACY_GL_EMULATION`, coming from an old commit https://github.com/kripken/emscripten/commit/d2a3ea8ca82533ac4dae07605aa56dee8bee9558.

**Comment by kripken** [comment]
Overall lgtm, but do we not expect chrome to fix this soon? (if soon, then maybe best to just wait?)

**Comment by gouletr** [comment]
@kripken without this fix we can't upgrade Emscripten otherwise it breaks running in Chrome release today.

**Comment by kripken** [comment]
I see, ok.

**Comment by juj** [comment]
> Overall lgtm, but do we not expect chrome to fix this soon? (if soon, then maybe best to just wait?)

They said they will have the fix in Chrome 58, and they're asking to land this workaround in the meanwhile and not wait. So I'll go ahead and push this in.

Thanks @gouletr for testing!

**Pull request merged**